File: Entities/TrackLocation.cs
namespace weatherapp.Entities;

//Entity is used to track the preferred user weather locations
public class TrackLocation: BaseEntity
{
	public string UserId  { get; set; }
	public Guid LocationId { get; set; }
	public bool isFavorite { get; set; } = false;
	public string? DisplayName { get; set; }
	
	public Location Location { get; set; }
}

File: frontend/app/types/trackLocation.ts
// Track location types and interfaces

export interface TrackLocation {
  id: string;
  isFavorite: boolean;
  displayName: string;
}

export interface CreateTrackLocationRequest {
  locationId: string;
  isFavorite?: boolean;
  displayName?: string;
}

export interface UpdateTrackLocationRequest {
  isFavorite?: boolean;
  displayName?: string;
}
